scenes/game/local_game.gd

```diff
@@ -2763,6 +2847,17 @@ func is_effect_condition_met(performing_player : Player, effect, local_condition
 		elif condition == "range":
 			var amount = effect['condition_amount']
 			var distance = abs(performing_player.arena_location - other_player.arena_location)
+
+			var from_source = 'condition_from_source' in effect and effect['condition_from_source']
```

Comment:
shouldn't need any of this if you just use litchi's thing, buddy_in_opponent_space